### ISSUE TITLE
Using bind(KProperty) in uninitialised ViewModel creates all types as ObjectProperty instead of IntegerProperty etc

### ISSUE THREAD

**Issue by dominic-jones** [edvin/tornadofx#365]
```
import javafx.application.Application
import javafx.beans.property.IntegerProperty
import javafx.beans.property.SimpleIntegerProperty
import javafx.scene.layout.HBox
import javafx.scene.layout.VBox
import tornadofx.*

class BasicApplication : App(BasicView::class)

fun main(args: Array<String>) {
    Application.launch(BasicApplication::class.java, *args)
}

class BasicView : View() {

    override val root = VBox()

    val model: PersonModel by inject() // not yet initialised

    init {
        root += find<BasicFragment>(mapOf(BasicFragment::int to model.int))
    }
}

class BasicFragment : Fragment() {

    val int: IntegerProperty by param()

    override var root = HBox()

    init {
        root = hbox {
            label(int)
        }
    }
}

class PersonModel : ItemViewModel<Person>() {
    val int: IntegerProperty = bind(Person::intProperty)

}

class Person(int: Int) {

    val intProperty = SimpleIntegerProperty(this, "int", int)
    var int by intProperty
}
```

There is a null handling block in ViewModel for this situation, looks something like this 
```
                null -> {
                    if (IntegerProperty::class.java.isAssignableFrom(PropertyType::class.java))
                        facade = BindingAwareSimpleIntegerProperty(this, null)
                    else if (DoubleProperty::class.java.isAssignableFrom(PropertyType::class.java))
                        facade = BindingAwareSimpleDoubleProperty(this, null)
                    else if (FloatProperty::class.java.isAssignableFrom(PropertyType::class.java))
                        facade = BindingAwareSimpleFloatProperty(this, null)
                    else if (BooleanProperty::class.java.isAssignableFrom(PropertyType::class.java))
                        facade = BindingAwareSimpleBooleanProperty(this, null)
                }
            }
```

PropertyType is reified in this method, but not in the caller of this method. Every time I ran through this method with the above example PropertyType was always ```javafx.beans.property.Property``` so all those paths fail.

As an aside, there's also no LongProperty branch here.

**Comment by edvin** [comment]
Thanks for reporting! Things are a bit slow moving due to vacation these days, will get back to you as fast as I can unless somebody beats me to it :)

**Comment by edvin** [comment]
I took a quick peek and it seems I was able to specify the property type as reified as well by changing the signature like so:

inline fun <reified N : Any, reified PropertyType: Property<N>, ReturnType : PropertyType> bind(property: KProperty1<T, PropertyType>, autocommit: Boolean = false, forceObjectProperty: Boolean = false): ReturnType

I will commit a fix shortly, just need to update all the other similar functions and run the test suite.

**Comment by edvin** [comment]
I've deployed a new snapshot (1.7.9-SNAPSHOT) to https://oss.sonatype.org/content/repositories/snapshots/no/tornado/tornadofx/, can you try it to see if it solves the problem for you?

**Comment by dominic-jones** [comment]
I've given this a go to no effect. I'm a bit loaded right now so haven't had a deeper dive into why this is a problem. From my own hacking in the past, I could get the reified property class in the first bind method, but not the second (the one with the if blocks). As I say, I need to debug a bit deeper into it.

Also, there's lots of info on best practices I'd like to pick up on in using this framework. There's a lot of potential here. Where's the link to the Slack channel you mentioned? Chatting through these issues would make a resolution faster

**Comment by edvin** [comment]
That's weird, I ran your exact code and it didn't work before, but worked after I reified the property type parameter. Let's discuss it further on slack :)

https://kotlinlang.slack.com/messages/tornadofx/

**Comment by edvin** [comment]
I have just released 1.7.9 - could you try again with that version?

**Comment by edvin** [comment]
This seems to be working, so I'll close the issue. Reopen if it still doesn't work as expected for you.
